contrib/segyconverter/utils/normalize_cube.py

```diff
@@ -0,0 +1,150 @@
+             The value was: {2}'.format(min_range, max_range, v))
+    return v
+
+
+def clip_file(local_filename, stddev, mean, k, min_range_value, max_range_value, normalize=True):
+    """
+        Normalize npy array file according to statistics. This function overwrites the existing
+        data file
+        :param str local_filename: npy data file
+        :param float stddev: standard deviation value
+        :param int k: number of standard deviation to be used in normalization
+        :param float min_range_value: minium range value
+        :param float max_range_value: maximum range value
+    """
+    # Load local npy file
+    cube = np.load(local_filename)
+    # Normalize block
+    try:
+        norm_cube = main(cube=cube, stddev=stddev, mean=mean, k=k, min_range=min_range_value,
+                         max_range=max_range_value, normalize=normalize)
+    except Exception as ex:
+        print("ERROR: Not possible to normalize cube. {}".format(ex))
+    # Save normalized cube locally
+    np.save(local_filename, norm_cube)
+
+
+def normalize_cube(cube: np.array, min_clip: float, max_clip: float, scale: float,
+                   min_range: float, max_range: float):
+    """
+        Normalize cube according to statistics
+        :param cube: 3D array to be normalized
+        :param min_clip: minimum value used for clipping
+        :param max_clip: maximum value used for clipping
+        :param min_range: minium range value
+        :param max_range: maximum range value
+        :param scale: scale value to be used for normalization
+        :returns: normalized 3D array
+        :rtype: numpy array
+    """
+    # Define function for normalization
+    vfunc = np.vectorize(norm_value)
+    # Normalize cube
+    norm_cube = vfunc(cube, min_clip=min_clip, max_clip=max_clip, min_range=min_range, max_range=max_range,
+                      scale=scale)
+    return norm_cube
+
+
+def main(cube: np.array, stddev: float, mean: float, k: float, min_range: float, max_range: float, normalize=True):
+    """
+    Compute statistics and normalize cube
+    :param cube: 3D array to be normalized
+    :param stddev: standard deviation value
+    :param k: number of standard deviation to be used in normalization
+    :param min_range: minium range value
+    :param max_range: maximum range value
+    :returns: normalized 3D array
+    :rtype: numpy array
+    """
+    if np.isnan(np.min(cube)):
+        raise Exception("Cube has NaN value")
+    if stddev == 0.0:
+        raise Exception("Standard deviation must not be zero")
+    if k == 0:
+        raise Exception("k must not be zero")
+
+    # Compute statistics
+    min_clip, max_clip, scale = compute_statistics(stddev=stddev, mean=mean, k=k, max_range=max_range)
+    if not normalize:
+        scale = 1
+    # Normalize cube
+    norm_cube = normalize_cube(cube=cube, min_clip=min_clip, max_clip=max_clip, scale=scale, min_range=min_range,
```

Comment:
Could we have two separate methods? One for normalization and other for clipping, instead of set scale = 1.